apps/client/src/features/app-settings/panel/general-panel/StyleEditor.tsx

```diff
@@ -0,0 +1,57 @@
+import { memo, MutableRefObject, useEffect, useState } from 'react';
+import Editor from 'react-simple-code-editor';
+import Prism from 'prismjs/components/prism-core';
+
+import 'prismjs/components/prism-css';
+import 'prismjs/themes/prism-tomorrow.min.css';
+import style from './StyleEditor.module.scss';
+
+interface CodeEditorProps {
+  language: string;
+  initialValue: string;
+  cssRef?: MutableRefObject<string>;
+}
+
+function CodeEditor(props: CodeEditorProps) {
+  const { language, initialValue, cssRef } = props;
+
+  const [code, setCode] = useState(initialValue);
+
+  useEffect(() => {
+    setCode(initialValue);
+  }, [initialValue]);
+
+  useEffect(() => {
```

Comment:
cleanup: can you please write a comment to what these useEffects achieve? like

`sync editor contents to external source`
and
`add contents to editor on mount`